### DISCUSSION TITLE
Vs code reports loop variable being captured by func literal with go1.22

### DISCUSSION THREAD

**Discussion by emar-kar** [golang/vscode-go#3381] [Help] [answered]
I have go1.22 installed and as far as I can see, there are no reports from vet, when I run it manually. But vs code itself still reports a warning for a loop variable being captured by func literal (loopclosure). Is it something I should do on my end in settings or is it a bug?

**Comment by emar-kar** [comment]
```go
package main

import (
	"fmt"
	"sync"
)

const num = 10

func main() {
	var wg sync.WaitGroup

	wg.Add(num)

	for i := range num {
		go func() {
			defer wg.Done()

			fmt.Println(i)
		}()
	}

	wg.Wait()
}
```

Example to reproduce.

**Comment by emar-kar** [comment]
![image](https://github.com/golang/vscode-go/assets/46078689/fe75fab9-8cc3-442d-a574-cf74a46dfce3)

I noticed that this behaviour changes as soon as I remove goroutine:
![image](https://github.com/golang/vscode-go/assets/46078689/82819867-9506-4d34-a11f-b886aa2f48f9)

**Comment by YusukeShimizu** [comment, marked as answer]
Probably because of this.
https://github.com/golang/go/issues/66876
